plugins/package-managers/node/src/main/kotlin/PackageJson.kt

```diff
@@ -141,11 +145,21 @@ data class PackageJson(
 
 private object AuthorListSerializer : JsonTransformingSerializer<List<Author>>(serializer<List<Author>>()) {
     override fun transformDeserialize(element: JsonElement): JsonElement =
-        // TODO: The string from the JSON primitive could be parsed into the dedicated author properties.
         when (element) {
             is JsonObject -> JsonArray(listOf(element))
-            is JsonPrimitive -> JsonArray(listOf(element.wrapPrimitiveInObject("name")))
-            is JsonArray -> JsonArray(element.map { it.wrapPrimitiveInObject("name") })
+            is JsonPrimitive -> JsonArray(listOf(element.toAuthorObject()))
+            is JsonArray -> JsonArray(element.map { it.toAuthorObject() })
+        }
+
+    private fun JsonElement.toAuthorObject(): JsonElement =
+        when (this) {
+            is JsonObject -> this
+            is JsonPrimitive -> {
+                val (name, email, url) = parseAuthorString(jsonPrimitive.contentOrNull)
+                val author = Author(name.orEmpty(), email, url)
```

Comment:
In case not all attributes can be extracte because the pattern does not match, shouldn't we just use the string content of the primitive as the name? e.g. the `orEmpty()` looks like that isn't the case.

Reply:
The code for this is now specifc to Node. However, as the previous commit statet that this is a common thing many package managers, should we change the behavior of that function?

I wonder if the following makes sense:

If the given string does not match the pattern, then return an Author object with only the name set, which includes the entire string.

Reply:
> The code for this is now specifc to Node.

Please explain why you think so. Esp., what has changed compared the previous iteration of his PR?

> If the given string does not match the pattern, then return an Author object with only the name set, which includes the entire string.

That's already the case. I've add a test for that to clarify.

Reply:
> lease explain why you think so. Esp., 

Consistency, prev commit introduces this function for consistency. But next one uses it in a special way.

> what has changed in this regard compared to the previous iteration of his PR?

Where is this question aiming at?

> That's already the case. I've add a test for that to clarify.

The last thing I've seen was this `?: content` code..

Reply:
It was not my intention to remove the fallback to the whole string as the author name if parsing fails. It should still be present.